web/blog/mobilizing-zeek-logs/index.md

```diff
@@ -0,0 +1,212 @@
+```
+
+And now you get JSON logs in the current directory.
+
+Because all log types are squashed into a single output stream, the package adds
+a new field `_path` to identify the log type. For example, the old `http.log` is
+now a JSON object with an additional field `{"_path": "http" , ...}`. Here's the
+same `conn.log` example from above, this time as NDJSON:
+
+```json
+{"_path":"conn","_write_ts":"2009-11-18T16:45:06.678526Z","ts":"2009-11-18T16:43:56.223671Z","uid":"CzFMRp2difzeGYponk","id.orig_h":"192.168.1.104","id.orig_p":1387,"id.resp_h":"74.125.164.85","id.resp_p":80,"proto":"tcp","service":"http","duration":65.45066595077515,"orig_bytes":694,"resp_bytes":11708,"conn_state":"SF","missed_bytes":0,"history":"ShADadfF","orig_pkts":9,"orig_ip_bytes":1062,"resp_pkts":14,"resp_ip_bytes":12276}
+{"_path":"conn","_write_ts":"2009-11-18T16:45:11.786059Z","ts":"2009-11-18T16:44:37.280644Z","uid":"CoalykVnjXndRUGP9","id.orig_h":"192.168.1.104","id.orig_p":1389,"id.resp_h":"67.195.146.230","id.resp_p":80,"proto":"tcp","service":"http","duration":29.393054962158203,"orig_bytes":791,"resp_bytes":38971,"conn_state":"SF","missed_bytes":0,"history":"ShADadfF","orig_pkts":14,"orig_ip_bytes":1359,"resp_pkts":35,"resp_ip_bytes":40379}
+{"_path":"conn","_write_ts":"2009-11-18T16:46:11.952479Z","ts":"2009-11-18T16:45:01.674828Z","uid":"Cz6QmkUOAQNigU7Ve","id.orig_h":"192.168.1.104","id.orig_p":1423,"id.resp_h":"74.125.164.85","id.resp_p":80,"proto":"tcp","service":"http","duration":65.16959500312805,"orig_bytes":3467,"resp_bytes":60310,"conn_state":"SF","missed_bytes":0,"history":"ShADadfF","orig_pkts":21,"orig_ip_bytes":4315,"resp_pkts":54,"resp_ip_bytes":62478}
+```
+
+A lot of tools have much less trouble with consuming such a single stream. That
+said, the JSON format is "dumbed down" compared to TSV, which contains
+additional type information, such as timestamps, durations, IP addresses, etc.
+This type information is lost in the JSON output and up to the downstream
+tooling to bring back.
+
+To read JSON Streaming Logs, use the [`zeek-json`](/next/formats/zeek-json)
+parser:
+
+```bash
+cat *.log | tenzir 'read zeek-json | taste 1'
+```
+
+The `zeek-json` parser just an alias for `json --selector=zeek:_path`, i.e.,
+Tenzir extracts the schema name from the `_path` field to demultiplex the JSON
+stream. As with any JSON, Tenzir's parser infers the richer types from strings,
+such as IP addresses and timestamps. Some information is lost, though, e.g., the
+fact that durations are simply floating point numbers.
+
+### Writer Plugin
+
+If the stock options of Zeek's logging framework do not work for you, you can
+still write a C++ *Writer Plugins* to produce any output of your choice.
+
+For example, the [zeek-kafka](https://github.com/SeisoLLC/zeek-kafka) plugin
+connects to a Kafka instance and writes JSON logs to specific topics. That would
+require a custom Zeek build, though. For this specific example, you could as
```

Comment:
> require a custom Zeek build

Not quite, with plugins you don't necessarily need to build Zeek yourself as well.

Reply:
Ah, right. From [the docs](https://docs.zeek.org/en/master/devel/plugins.html):

> The configure script will just need to be able to find the location of either a Zeek installation-tree or a Zeek source-tree.